Error in `ColExpressionsTest::test_group_by` test when using Postgres 16 Beta 1

### Describe the bug

Since c postgres 16beta1 the order of rows after grouping has changed

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

_No response_

### SQLAlchemy Version in Use

main

### DBAPI (i.e. the database driver)

asyncpg

### Database Vendor and Major Version

PostgreSQL 16 Beta 1

### Python Version

3.11.2

### Operating system

OSX

### To Reproduce

```python
pytest --no-header --tb=no --dburi=postgresql+asyncpg://test:test@localhost:5432/test test/orm/inheritance/test_basic.py::ColExpressionsTest::test_group_by
```


### Error

postgres:16beta1-alpine3.18
```
FAILED test/orm/inheritance/test_basic.py::ColExpressionsTest_postgresql+asyncpg_async_16::test_group_by - AssertionError: [(2, 2, 7), (1, 1, 5)] != [(1, 1, 5), (2, 2, 7)]
```

postgres:15.4-alpine3.18
```
test/orm/inheritance/test_basic.py::ColExpressionsTest_postgresql+asyncpg_async_15_4::test_group_by PASSED
```

--- a/test/orm/inheritance/test_basic.py
+++ b/test/orm/inheritance/test_basic.py
@@ -177,6 +177,7 @@ class ColExpressionsTest(fixtures.DeclarativeMappedTest):
         rows = (
             s.query(B.id.expressions[0], B.id.expressions[1], func.sum(B.data))
             .group_by(*B.id.expressions)
+            .order_by(B.id.expressions[0])
             .all()
         )
         eq_(rows, [(1, 1, 5), (2, 2, 7)])

